web/src/channel_folders_popover.ts

```diff
@@ -0,0 +1,68 @@
+import $ from "jquery";
+import assert from "minimalistic-assert";
+import type * as tippy from "tippy.js";
+
+import render_left_sidebar_channels_folder_setting_popover from "../templates/popovers/left_sidebar/left_sidebar_channels_folder_setting_popover.hbs";
+
+import * as channel from "./channel.ts";
+import * as popover_menus from "./popover_menus.ts";
+import {parse_html} from "./ui_util.ts";
+import {user_settings} from "./user_settings.ts";
+
+function register_toggle_channel_folders(
```

Comment:
I don't think `register` is part of the description of this function? `do_change_show_channel_folders` seems like a much better name.

Also, maybe the caller can just pass `instance` ? Just feels like a weird abstraction to be parsing data out of the `event`, when the caller already had `instance` available.